### PR TITLE
Update compensation of aperture and ISO change in HDR fusion

### PR DESCRIPTION
**Pull request by servantf**
HDR fusion did not work when using multiple brackets with variable ISO and/or Aperture.
This PR updates the equations used.

### CODE REVIEW

File: src/aliceVision/sfmData/View.cpp [hunk 1]
```diff
@@ -17,18 +17,45 @@
...
 namespace sfmData {
 
-float View::getCameraExposureSetting() const
+float View::getCameraExposureSetting(const float referenceISO, const float referenceFNumber) const
 {
     const float shutter = getMetadataShutter();
     const float fnumber = getMetadataFNumber();
+    const float iso = getMetadataISO();
+    
     if(shutter < 0 || fnumber < 0)
```
Comment (fabiencastan) [inline src/aliceVision/sfmData/View.cpp:26]: if iso < 0

File: src/aliceVision/sfmData/View.cpp [hunk 2]
```diff
@@ -17,18 +17,45 @@
...
-    float cameraExposure = (shutter * isoRatio) / (fnumber * fnumber);
-    return cameraExposure;
+    float lReferenceFNumber = referenceFNumber;
+    if (lReferenceFNumber < 0.0f) {
+        lReferenceFNumber = fnumber;
+    }
+    
+    /* 
+    iso = qLt / aperture^2
+    isoratio = iso2 / iso1 = qLt / aperture1² / (qLt / aperture2²)
```
Comment (fabiencastan) [inline src/aliceVision/sfmData/View.cpp:41]: use `^2` instead of special characters

File: src/aliceVision/sfmData/View.cpp [hunk 3]
```diff
@@ -17,18 +17,45 @@
...
+    const float iso = getMetadataISO();
+    
     if(shutter < 0 || fnumber < 0)
         return -1.f;
 
-    const float iso = getMetadataISO();
-    const float isoRatio = (iso < 0.f) ? 1.0 : (iso / 100.f);
+    float lReferenceIso = referenceISO;
+    if (lReferenceIso < 0.0f) {
+        lReferenceIso = iso;
```
Comment (fabiencastan) [inline src/aliceVision/sfmData/View.cpp:31]: It does not make sense to be relative to itself.

File: src/aliceVision/sfmData/View.cpp (hunk 2 above)
Comment (servantf) [reply src/aliceVision/sfmData/View.cpp:41]: Ok

File: src/software/pipeline/main_LdrToHdrCalibration.cpp [hunk 4]
```diff
@@ -353,8 +353,9 @@ int aliceVision_main(int argc, char** argv)
         std::vector<float> exposures;
 
         for(int j = 0; j < group.size(); ++j)
-        {
-            float etime = group[j]->getCameraExposureSetting();
+        {   
+            
+            float etime = group[j]->getCameraExposureSetting(group[0]->getMetadataISO(), group[0]->getMetadataFNumber());
```
Comment (fabiencastan) [inline src/software/pipeline/main_LdrToHdrCalibration.cpp:358]: Could/Should we use the default values here too?

Comment (servantf) [reply src/software/pipeline/main_LdrToHdrCalibration.cpp:358]: :-/

File: src/aliceVision/sfmData/View.cpp (hunk 3 above)
Comment (fabiencastan) [reply src/aliceVision/sfmData/View.cpp:31]: Resolve as the default value is no more -1.

File: src/aliceVision/sfmData/View.cpp [hunk 5]
```diff
@@ -17,18 +17,45 @@
...
     const float fnumber = getMetadataFNumber();
+    const float iso = getMetadataISO();
+    
     if(shutter < 0 || fnumber < 0)
         return -1.f;
 
-    const float iso = getMetadataISO();
-    const float isoRatio = (iso < 0.f) ? 1.0 : (iso / 100.f);
+    float lReferenceIso = referenceISO;
+    if (lReferenceIso < 0.0f) {
```
Comment (fabiencastan) [inline src/aliceVision/sfmData/View.cpp:30]: We should do something is "iso" is -1.
If "iso" or "referenceISO" or both are -1, we should put both to the same value to neutralize it.

**Review by fabiencastan — APPROVED** [review]

**Pull request merged**